jiant/models.py

```diff
@@ -1160,6 +1174,69 @@ def _lm_forward(self, batch, task, predict):
             pass
         return out
 
+    def _masked_lm_forward(self, batch, task, predict):
+        mlm_probability = 0.15
+        out = {}
+        sent_encoder = self.sent_encoder
+        tokenizer_name = self.sent_encoder._text_field_embedder.input_module
+        vocab_size = (
+            self.sent_encoder._text_field_embedder.model.embeddings.word_embeddings.num_embeddings
+        )
+        tokenizer = get_tokenizer(tokenizer_name)
+        input_key = self.sent_encoder._text_field_embedder.tokenizer_required
+        # mask_idx = self.sent_encoder._text_field_embedder._mask_id #
+        mask_idx = self.sent_encoder._text_field_embedder._mask_id
+        b_size, seq_len = batch["targs"].size()
+        inputs = batch["input"][input_key]
+        labels = batch["targs"]
+
+        probability_matrix = torch.full(labels.shape, mlm_probability, device=inputs.device)
+        padding_mask = labels.eq(0)
+        probability_matrix.masked_fill_(padding_mask, value=0.0)
+
+        masked_indices = torch.bernoulli(probability_matrix).to(
+            device=inputs.device, dtype=torch.uint8
+        )
+
+        labels[~masked_indices] = -100  # We only compute loss on masked tokens
```

Comment:
Why is this needed, then?

Reply:
@phu-pmh can explain this better than I.

Reply:
nn.CrossEntropy ignores the idices with value = -100 by default. Therefore, we replace non-masked indices with -100 not to consider them in loss computation.

Reply:
Thanks for the info @phu-pmh. Requesting that this info gets captured here with a comment in the code.